src/services/zapper.js

```diff
@@ -0,0 +1,228 @@
+// @flow
+/*
+    Pillar Wallet: the personal data locker
+    Copyright (C) 2021 Stiftung Pillar Project
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+/* eslint-disable i18next/no-literal-string */
+
+// utils
+import httpRequest from 'utils/httpRequest';
+import { reportErrorLog } from 'utils/common';
+
+// constants
+import { CHAIN } from 'models/Chain';
+import { PROTOCOLS } from 'constants/zapperServiceConstants';
+
+// types
+import type { Chain } from 'models/Chain';
+
+// does not change between envs
+const ZAPPER_CONFIG = {
+  API_URL: 'https://api.zapper.fi/v1',
+  API_KEY: '96e0cc51-a62e-42ca-acee-910ea7d2a241', // public
+};
+
+const requestConfig = {
+  timeout: 10000,
+  headers: {
+    Accept: 'application/json',
+    'Content-Type': 'application/json',
+  },
+};
+
+const buildAddressesQuery = (
+  addresses: string[],
+) => `addresses[]=${addresses.map((address) => `${address}`).join('&addresses[]=')}`;
+
+const mapZapperNetworkIdToChain = (network: string): ?Chain => {
+  switch (network) {
+    case 'ethereum': return CHAIN.ETHEREUM;
+    case 'polygon': return CHAIN.POLYGON;
+    case 'binance-smart-chain': return CHAIN.BINANCE;
+    case 'xdai': return CHAIN.XDAI;
+    default: return null;
+  }
+};
+
+
+const walletProtocols = [
+  PROTOCOLS.TOKENS,
+  PROTOCOLS.BITCOIN,
```

Comment:
Note sure TBH, @IAmKio added protocol values from Zapper Swagger. However, we also had a discussion earlier today and agreed that wallet balances will be going from Etherspot only because Zapper does not support xDai balances and Etherspot can be more trusted on actual balances when it comes to sending right asset amounts. We leave as it is now and will revert back to Etherspot in separate PR.